src/main/java/antlr/treeFormatParser/TreeFormatVisitorImplementation.kt

```diff
@@ -0,0 +1,66 @@
+package antlr.treeFormatParser
+
+
+import antlr.treeFormatParser.generated.TreeFormatParser
+import antlr.treeFormatParser.generated.TreeFormatParser.InformationContext
+import antlr.treeFormatParser.generated.TreeFormatVisitor
+import org.antlr.v4.runtime.tree.ErrorNode
+import org.antlr.v4.runtime.tree.ParseTree
+import org.antlr.v4.runtime.tree.RuleNode
+import org.antlr.v4.runtime.tree.TerminalNode
+import tree.ImpTargetTreeNode
+import tree.TargetTreeNode
+
+class TreeFormatVisitorImplementation : TreeFormatVisitor<TargetTreeNode> {
+
+    var idCount: Int = 0
+
+    override fun visitSubtree(ctx: TreeFormatParser.SubtreeContext): TargetTreeNode {
+        var node: ImpTargetTreeNode = visitNode(ctx.node()) as ImpTargetTreeNode
+        if(ctx.child() == null){
+            return node
+        }
+
+        node.setChildren(visitChild(ctx.child()).getChildren()!!)
+        return node
+    }
+
+    override fun visitChild(ctx: TreeFormatParser.ChildContext): TargetTreeNode {
+        return visitSibling(ctx.sibling())
+    }
+
+    override fun visitSibling(ctx: TreeFormatParser.SiblingContext): TargetTreeNode {
+        val siblings = ctx.subtree().toList()
+        var siblingNodes = ArrayList<TargetTreeNode>()
+        for(subtree in siblings){
+            siblingNodes.add(visitSubtree(subtree))
+        }
+        var aux = ImpTargetTreeNode("", "", -1, null, null)
+        aux.setChildren(siblingNodes)
+        return aux
+    }
+
+    override fun visitNode(ctx: TreeFormatParser.NodeContext): TargetTreeNode {
+        return ImpTargetTreeNode(ctx.name.text, ctx.tag.text, idCount++, null, null)
```

Comment:
It is necessary to put null , null in the constructor, change this for a default value.